### DISCUSSION TITLE
How to load zip contents in a treeview

### DISCUSSION THREAD

**Discussion by gh658804** [icsharpcode/SharpZipLib#563] [Q&A]
Hi.

In a personal project, i would like to load all contents of a zip file (the names only of course) in a treeview in order to see the zip structure there. I can't decide how to proceed though. From what i've found in the net, the "standard" way to parse a zip file is with a "foreach" loop, however i have found no way to associate a folder ZipEntry with its children (files/folders ZipEntries in it), apart from splitting its full name and checking that against other ZipEntries.

Has anyone found (or created) code for performing such a task or should i try to invent the wheel? It would at least help to know if the ZipEntries are somehow sorted and child ZipEntries will never appear before their respective parents in a "foreach" loop.

**Comment by piksel** [comment]
The code for building the treeview in Dearchiver can be found here:
https://github.com/piksel/Dearchiver/blob/main/src/Extractor.cs

It's basically doing that. There are no guarantees of the order afaik. It is really slow on large archives though, so it tends to just bail on these and show a reduced tree view. Especially since I rarely even use the tree.

**Reply by gh658804** [reply]
Thanks for that. I'll check it out.
